### PR TITLE
Prepend SyncFileManager path with `app_id`

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

File: tests/sync/sync_manager.cpp [hunk 1]
```diff
@@ -85,34 +85,34 @@ TEST_CASE("sync_manager: `path_for_realm` API", "[sync]") {
 
     SECTION("should work properly without metadata") {
         TestSyncManager init_sync_manager({.base_path = base_path, .metadata_mode = SyncManager::MetadataMode::NoMetadata});
-        const auto expected = base_path + "mongodb-realm/app_id/foobarbaz/realms%3A%2F%2Frealm.example.org%2Fa%2Fb%2F%7E%2F123456%2Fxyz.realm";
+        const auto expected = base_path + "app_id/mongodb-realm/foobarbaz/realms%3A%2F%2Frealm.example.org%2Fa%2Fb%2F%7E%2F123456%2Fxyz.realm";
```
Comment (cmelchior) [inline tests/sync/sync_manager.cpp:88]: I'm a bit uncomfortable doing it like this with the app_id as a prefix. At least on Android files are being stored in a shared files folder, which means we will potentially create many folders that might conflict with already existing files or folders. I would much prefer if we only had `mongodb-realm` and then something like `mongodb-realm/<appId>/server-utility` and `mongodb-realm/<appId>/recovered-realms`

**Review by cmelchior — CHANGES_REQUESTED** [review]

**Comment by jsflax** [comment]
@cmelchior PTAL

File: src/sync/impl/sync_file.hpp [hunk 2]
```diff
@@ -111,8 +107,12 @@ class SyncFileManager {
     }
 
 private:
+    // Denotes the root path for any mongodb-realm app for the passed in `base_path`.
+    // Expected to be `base_path` + "mongodb-realm/".
+    const std::string m_root_sync_path;
+    // Denotes the base path for the mongodb-realm app associated with this sync manager.
```
Comment (cmelchior) [inline src/sync/impl/sync_file.hpp:113]: Maybe it is just me, but the wording of these comments seems to indicate the naming of the variables should be switched?

File: tests/sync/file.cpp [hunk 3]
```diff
@@ -148,15 +148,15 @@ TEST_CASE("sync_file: URL manipulation APIs", "[sync]") {
...
     const std::string identity = "abcdefghi";
     const std::string local_identity = "123456789";
-    const std::string manager_path = base_path + "syncmanager/";
     const std::string app_id = "test_app_id*$#@!%1";
     const std::string expected_clean_app_id = "test_app_id%2A%24%23%40%21%251";
+    const std::string manager_path = base_path + "syncmanager/mongodb-realm/" + expected_clean_app_id + "/";
     prepare_sync_manager_test();
     auto cleanup = util::make_scope_exit([=]() noexcept { util::try_remove_dir_recursive(base_path); });
-    auto manager = SyncFileManager(manager_path, app_id);
+    auto manager = SyncFileManager(base_path + "syncmanager/", app_id);
```
Comment (cmelchior) [inline tests/sync/file.cpp:156]: Minor, but not 100% sure why we need the `syncmanager` subfolder here?

**Review by cmelchior — APPROVED** [review]
Nice 💯 . Only had a few cosmetic comments.

File: src/sync/impl/sync_file.hpp (hunk 2 above)
Comment (jsflax) [reply src/sync/impl/sync_file.hpp:113]: Hmm... I think the `base_path` parameter in the constructor makes in confusing. You are probably correct, but it does mean `m_base_path` becomes moot.

File: tests/sync/file.cpp (hunk 3 above)
Comment (jsflax) [reply tests/sync/file.cpp:156]: I mainly converted this test to make sure it worked as it did previously– my guess is that this is something legacy-ish.

**Pull request merged**